### ISSUE TITLE
SamplePulse drawer returns two figures in Jupyter 

### ISSUE THREAD

**Issue by nonhermitian** [Qiskit/qiskit-terra#2449]
### Information

- **Qiskit Terra version**: master
- **Python version**:
- **Operating system**:

### What is the current behavior?
```python
import numpy as np
from qiskit import *
from qiskit.pulse import *
x_data = np.array([0.004, 0.029, 0.135, 0.41, 0.8, 1.0, 0.8, 0.41, 0.135, 0.029, 0.004])
XPulse = SamplePulse(x_data)
XPulse.draw()
```
gives:

<img width="463" alt="Screen Shot 2019-05-17 at 2 32 49 PM" src="https://user-images.githubusercontent.com/1249193/57948819-ae93a680-78b0-11e9-80de-fd184671d82c.png">

### Suggested solutions
The figure instance needs to be closed before returning.

**Comment by nonhermitian** [comment]
Same is true for schedule drawer.

**Comment by taalexander** [comment]
Can you look at this @nkanazawa1989?

**Comment by raffmiceli** [comment]
I can recreate the behavior in Jupyter Notebook by adding %matplotlib inline after the import statements, but I can't get the figure to pop up in a new window when I run the code from the console. However, it works as expected from the console if I add a filename as an argument to draw(), and the file contains only one copy of the plot.

I'm running this through Windows Subsystem for Linux on Windows 10.

**Comment by nkanazawa1989** [comment]
@taalexander Ok, I'll check.

**Comment by nkanazawa1989** [comment]
@raffmiceli 
> but I can't get the figure to pop up in a new window when I run the code from the console

please try `interactive=True` when drawing the figure from the console

**Comment by taalexander** [comment]
Closed by #2515.
